src/NodeVisitor/EscaperNodeVisitor.php

```diff
@@ -82,6 +90,39 @@ protected function doLeaveNode(Node $node, Environment $env)
         return $node;
     }
 
+    private function unwrapConditional(ConditionalExpression $expression, Environment $env)
+    {
+        // convert "echo a ? b : c" to "a ? echo b : echo c" recursively
+        $expr2 = $expression->getNode('expr2');
+        if ($expr2 instanceof ConditionalExpression) {
+            $expr2 = $this->unwrapConditional($expr2, $env);
+        } else {
+            $expr2 = $this->escapeInlinePrintNode(new InlinePrint($expr2, $expr2->getTemplateLine()), $env, $this->needEscaping($env));
```

Comment:
why not passing the escaping type from the caller, as you already have it on line 75 (and you would then have it in this method for recursive calls), instead of re-checking it each time ?

Reply:
done